src/ast/irbuilderbpf.cpp

```diff
@@ -15,6 +15,20 @@ namespace libbpf {
 namespace bpftrace {
 namespace ast {
 
+StructType *IRBuilderBPF::GetStructType(
```

Comment:
What's the advantage of caching the StructTypes like this?

Reply:
The main advantage is that we don't define a lot of duplicate types in the  the generated ir (before it goes through llvm passes), making it  a bit easier to read. e.g. for  tools/tcpaccept.bt we generate a ton of structs:

```
%printf_t.1 = type { i64, [64 x i8], [64 x i8], [64 x i8], [64 x i8], [64 x i8] }
%printf_t.0 = type { i64, [64 x i8], [64 x i8], [64 x i8] }
%printf_t = type { i64 }
%printf_t.9 = type { i64, i64, i64 }
%printf_t.8 = type { i64, [24 x i8], i64, [24 x i8], i64 }
%printf_t.7 = type { i64, i64, [16 x i8] }
%inet_t.6 = type { i64, [16 x i8] }
%inet_t.5 = type { i64, [16 x i8] }
%inet_t.4 = type { i64, [16 x i8] }
%inet_t.3 = type { i64, [16 x i8] }
%inet_t.2 = type { i64, [16 x i8] }
%inet_t = type { i64, [16 x i8] }
```

And it might remove some extra work LLVM has to do de-duplicating all types.